### PR TITLE
Add verbose flag for the run command

### PR DESCRIPTION
**Pull request by johnpoth**
Thanks !

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by squakez — CHANGES_REQUESTED** [review]
We need to get rid off the `fmt.Print` default favouring `fmt.Fprintln(cmd.OutOrStdout(), ...)`. Even more, as it's a warning, I think we better use stderr via `cmd.ErrOrStderr()`.

Additionally I'd wrap a function, ie, `printOut(verbose bool)` where you have the `if` condition, and call directly this function within the rest of the code. We will have a more readable code.

File: pkg/cmd/run.go [hunk 1]
```diff
@@ -782,15 +788,15 @@ func resolvePodTemplate(ctx context.Context, cmd *cobra.Command, templateSrc str
 	return err
 }
 
-func uploadFileOrDirectory(platform *v1.IntegrationPlatform, item string, integrationName string, cmd *cobra.Command, integration *v1.Integration) error {
+func uploadFileOrDirectory(platform *v1.IntegrationPlatform, item string, integrationName string, cmd *cobra.Command, integration *v1.Integration, verbose bool) error {
```
Comment (tadayosi) [inline pkg/cmd/run.go:791]: Maybe we can change it into method of `runCmdOptions` as the args now looks a bit too long.

File: pkg/cmd/run.go [hunk 2]
```diff
@@ -850,7 +860,7 @@ func getMountPath(targetPath string, dirName string, path string) (string, error
 }
 
 // nolint:errcheck
-func uploadPomFromJar(gav maven.Dependency, path string, platform *v1.IntegrationPlatform, ns string, options spectrum.Options) maven.Dependency {
+func uploadPomFromJar(gav maven.Dependency, path string, platform *v1.IntegrationPlatform, ns string, options spectrum.Options, verbose bool) maven.Dependency {
```
Comment (tadayosi) [inline pkg/cmd/run.go:863]: Maybe we can change it into method of `runCmdOptions` as the args now looks a bit too long.

File: pkg/cmd/run.go [hunk 3]
```diff
@@ -945,14 +955,16 @@ func extractGav(src *zip.File, localPath string) (maven.Dependency, bool) {
 	}, true
 }
 
-func uploadAsMavenArtifact(dependency maven.Dependency, path string, platform *v1.IntegrationPlatform, ns string, options spectrum.Options) error {
+func uploadAsMavenArtifact(dependency maven.Dependency, path string, platform *v1.IntegrationPlatform, ns string, options spectrum.Options, verbose bool) error {
```
Comment (tadayosi) [inline pkg/cmd/run.go:958]: Maybe we can change it into method of `runCmdOptions` as the args now looks a bit too long.

File: pkg/cmd/run.go [hunk 4]
```diff
@@ -1028,15 +1040,19 @@ func writeChecksumToFile(filepath string, hash hash.Hash) error {
 	return err
 }
 
-func getSpectrumOptions(platform *v1.IntegrationPlatform, cmd *cobra.Command) spectrum.Options {
+func getSpectrumOptions(platform *v1.IntegrationPlatform, cmd *cobra.Command, verbose bool) spectrum.Options {
```
Comment (tadayosi) [inline pkg/cmd/run.go:1043]: Maybe we can change it into method of `runCmdOptions` as the args now looks a bit too long.

File: pkg/cmd/run.go [hunk 5]
```diff
@@ -117,6 +117,7 @@ func newCmdRun(rootCmdOptions *RootCmdOptions) (*cobra.Command, *runCmdOptions)
 	cmd.Flags().String("pod-template", "", "The path of the YAML file containing a PodSpec template to be used for the Integration pods")
 
 	cmd.Flags().Bool("save", false, "Save the run parameters into the default kamel configuration file (kamel-config.yaml)")
+	cmd.Flags().Bool("verbose", false, "Verbose logging")
```
Comment (tadayosi) [inline pkg/cmd/run.go:120]: Is there no shorthand name for the flag? :smiley:

It looks like the `verbose` flag can be global, but maybe we can work on it later?

**Review by tadayosi — APPROVED** [review]

**Comment by tadayosi** [comment]
> Additionally I'd wrap a function, ie, printOut(verbose bool) where you have the if condition, and call directly this function within the rest of the code. We will have a more readable code.

That sounds a nice idea. I'd push it further and make wrapped printing functions as methods for `rootCmdOptions` so that all child commands can exploit them. Maybe we could have `o.printOut()`, `o.printErr()`, `o.printVerboseOut()`, and `o.printVerboseErr()`.

File: pkg/cmd/run.go (hunk 5 above)
Comment (johnpoth) [reply pkg/cmd/run.go:120]: Ah yeah b/c it was already taken by the volume flag ... wasn't sure what to do in that case :) vb ?

Comment (tadayosi) [reply pkg/cmd/run.go:120]: Ah I see, we already have `--volume`... Maybe then we can use `-V`?
It's ok to leave it without a shorthand flag if we don't find a good one.

**Comment by johnpoth** [comment]
maybe worth switching to using *klog* like we do in the operator

**Comment by squakez** [comment]
> maybe worth switching to using _klog_ like we do in the operator

Sure. Alternatively we can finalize this and make a follow up issue to do it.

**Comment by johnpoth** [comment]
Sounds good ! Applied the changes, thanks !

File: pkg/cmd/run.go [hunk 6]
```diff
@@ -1032,15 +1032,19 @@ func writeChecksumToFile(filepath string, hash hash.Hash) error {
 	return err
 }
 
-func getSpectrumOptions(platform *v1.IntegrationPlatform, cmd *cobra.Command) spectrum.Options {
+func (o *runCmdOptions) getSpectrumOptions(platform *v1.IntegrationPlatform, cmd *cobra.Command) spectrum.Options {
 	insecure := platform.Spec.Build.Registry.Insecure
+	var stdout io.Writer
+	if o.Verbose {
+		stdout = cmd.OutOrStdout()
+	}
```
Comment (squakez) [inline pkg/cmd/run.go:1040]: What happen if `o.Verbose==false`? is it okey if we leave the `nil` or is it going to break in the Spectrum struct?

Comment (johnpoth) [reply pkg/cmd/run.go:1040]: Hi @squakez if it's false (default) then *stdout* stays nil and nothing will be sent to the console

**Review by squakez — APPROVED** [review]

**Comment by johnpoth** [comment]
Thanks @squakez @tadayosi for the review! merging ...

**Pull request merged**